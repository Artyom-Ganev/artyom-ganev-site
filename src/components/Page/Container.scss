.page-container {
  &__root {
    max-width: 50vw;

    &, &--narrow {
      margin: 0.8em;
      border-radius: 0.8em;
      padding: 0.8em;
      -moz-box-shadow: 0 0 2px 1px #CCCCCC;
      -webkit-box-shadow: 0 0 2px 1px #CCCCCC;
      box-shadow: 0 0 2px 1px #CCCCCC;
      opacity: 0.8;
      background-color: rgba(255, 255, 255, 0.8);
    }

    &--narrow {
      max-width: 95vw;
    }

    &:hover {
      background-color: rgba(255, 255, 255, 0.9);
    }
  }
}