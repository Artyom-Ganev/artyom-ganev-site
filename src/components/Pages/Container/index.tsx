import Box from '@material-ui/core/Box';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { LoadingIndicator } from 'components/Navigation';
import { IPageContainer } from 'components/Pages';
import React, { FC, PropsWithChildren } from 'react';
import { useSelector } from 'react-redux';
import { isLoading } from 'selectors';
import { IState } from 'types';
import useStyles from './styles';

/**
 * Page container with title
 */
const Container: FC<IPageContainer> = ({ children, Icon, title }: PropsWithChildren<IPageContainer>) => {
  const classes = useStyles();
  const loading = useSelector<IState, boolean>(isLoading);
  return (
    <Paper className={classes.pageContainer} square>
      {loading ? (
        <LoadingIndicator />
      ) : (
        <>
          <Box display='flex' alignItems='center'>
            {Icon && <Icon fontSize='large' className={classes.pageContainerIcon} />}
            {title && (
              <Typography variant='h5' className={classes.pageContainerTitle}>
                {title}
              </Typography>
            )}
          </Box>
          {children}
        </>
      )}
    </Paper>
  );
};

export default Container;
