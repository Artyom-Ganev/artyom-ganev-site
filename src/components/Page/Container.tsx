import React, { ReactNode } from 'react';
import { connect } from 'react-redux';
import IState from '../../interface/IState';
import * as actions from '../../actions';
import LoadingIndicator from '../Navigation/LoadingIndicator';
import './Container.scss';

/**
 * Container properties interface
 */
interface IProperties {
  title?: string;
  children?: ReactNode | ReactNode[];
  loading?: boolean;
  narrow: boolean;
}

const mapStateToProps = ({ app: { loading, narrow } }: IState) => ({ loading, narrow });

const actionCreators = {
  appLoading: actions.appLoading,
};

/**
 * Page container with title
 */
const Container = (props: IProperties) => {
  const { children, loading, narrow, title } = props;
  const contentClass = `flexBox flexColumn page-container__root${narrow ? '--narrow' : ''}`;
  return loading ? (
    <div className={contentClass}>
      <LoadingIndicator />
    </div>
  ) : (
    <div className={contentClass}>
      {title && <h2>{title}</h2>}
      {children}
    </div>
  );
};

export default connect(mapStateToProps, actionCreators)(Container);
