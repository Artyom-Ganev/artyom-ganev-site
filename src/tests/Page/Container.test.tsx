import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../../components/Page/Container';
import { combineReducers, createStore } from 'redux';
import { Provider } from 'react-redux';

const app = () => ({ loading: false });

const store = createStore(combineReducers({ app }));

test('Container', () => {
  const title = 'title 1';
  const content = <div>text1</div>;
  const component = renderer.create(
    <Provider store={store}>
      <Container title={title}>{content}</Container>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Container', () => {
  const content = <div>text1</div>;
  const component = renderer.create(
    <Provider store={store}>
      <Container>{content}</Container>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
