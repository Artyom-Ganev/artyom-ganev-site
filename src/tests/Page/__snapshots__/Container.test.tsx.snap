// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container 1`] = `
<div
  className="flexBox flexColumn page-container__root"
>
  <h2>
    title 1
  </h2>
  <div>
    <div>
      text1
    </div>
  </div>
</div>
`;

exports[`Container 2`] = `
<div
  className="flexBox flexColumn page-container__root"
>
  <div>
    <div>
      text1
    </div>
  </div>
</div>
`;